packages/client/src/game/Game.ts

```diff
@@ -0,0 +1,362 @@
+import { TableEntity } from './entities/TableEntity';
+import { EventBus } from './utils/EventBus';
+
+export class Game extends EventBus {
+  activePlayerId = -1;
+  players: GamePlayerType[] = [];
+  handEntities: Record<string, HandEntity> = {};
+  table!: TableEntity;
+
+  activeColor = '';
+  clockwiseMovement = true;
+
+  constructor() {
+    super();
+  }
+
+  startGame(playersNum: number, playerData: GamePlayerType) {
+    this.players = addPlayers(playersNum, playerData);
+
+    /* Перемешиваем массив карт. Это стартовая колода */
+    const initialPack = shuffle(allCards);
+
+    this.table = new TableEntity();
+    this.table.start();
+
+    /* Создаём "руки" игроков и раздаём карты */
+    let zIndex = 10;
+    for (let i = 0; i < playersNum; i++) {
+      const entityName =
+        playersNum === 2 ? TwoPlayerLayers[i] : FourPlayerLayers[i];
+      this.handEntities[entityName] = new HandEntity(
+        entityName as HandEntityTypes,
+        this.players[i]
+      );
+
+      const cards = initialPack.splice(
+        initialPack.length - 1 - START_NUM_CARDS_IN_HAND,
+        START_NUM_CARDS_IN_HAND
+      );
+      this.handEntities[entityName].create(zIndex);
+      this.handEntities[entityName].addCards(cards);
+
+      zIndex -= 1;
+    }
+
+    /* Из стартовой колоды берём одну карту для открытия */
+    const lastCard = initialPack.pop();
+    if (lastCard) {
+      /* Открываем карту поле раздачи всех карт игрокам */
+      setTimeout(() => {
+        this.table.addUpcard(lastCard);
+      }, ANIMATION_TIME * START_NUM_CARDS_IN_HAND);
+
+      if (lastCard.action) {
+        setTimeout(() => {
+          this.checkCard(lastCard);
+          this.moveLine();
+        }, ANIMATION_TIME * (START_NUM_CARDS_IN_HAND + 1));
+      } else {
+        this.activePlayerId = 0;
+        this.getActivePlayer().highlight();
+      }
+
+      this.activeColor = lastCard.color;
+    }
+    /* Оставшаяся часть карт будет в закрытой колоде */
+    this.table.setClosePack(initialPack);
+  }
+
+  /* Проверяем, может ли в текущий момент игрок выполнить ход */
+  checkPosiibilityOfAction() {
+    return this.activePlayerId === 0;
+  }
+
+  /* Юзер берёт карту со стола */
+  playerTakeCard(clickPos: Record<string, number>) {
+    const playerLayer = this.handEntities['frontHand'];
+    const playerCards = playerLayer.cards;
+    const { x: xClick, y: yClick } = clickPos;
+
+    /* Если клик по слою стола был не на закрытой колоде, ничего не происходит */
+    if (
+      xClick <= 0 &&
+      xClick >= BASE_WIDTH_CARD &&
+      yClick <= 0 &&
+      yClick >= BASE_HEIGHT_CARD
+    ) {
+      console.log('Чтобы взять карту, нужно кликнуть по закрытой колоде');
+      return;
+    }
+
+    /* Если есть карта, которую можно скинуть, то взять из закрытой колоды нельзя */
+    for (let i = playerCards.length - 1; i >= 0; i--) {
+      const card = playerCards[i];
+
+      if (compareCards(card, this.table.getUpcard(), this.activeColor)) {

```

Comment:
Для повышения читабельности кода, лучше такие сложные вызовы вывести в отдельную переменную, и проверят его
```suggestion
      const isCardsSomething = compareCards(card, this.table.getUpcard(), this.activeColor)
      if (isCardsSomething) {
```